app/src/store/actions/AlgorithmDialogStoreActions.ts

```diff
@@ -0,0 +1,128 @@
+import { Algorithm, AlgorithmApi } from 'api';
+import { AlgorithmDialogStoreActionTypes as DialogActions } from 'store/actions/actionTypes';
+import { getAlgorithms } from 'store/actions/AlgorithmsStoreActions';
+import { SnowmanDispatch, SnowmanThunkAction } from 'store/messages';
+import { store } from 'store/store';
+import { MagicNotPossibleId } from 'utils/constants';
+import {
+  easyPrimitiveAction,
+  easyPrimitiveActionReturn,
+} from 'utils/easyActionsFactory';
+import RequestHandler from 'utils/requestHandler';
+import {
+  SUCCESS_TO_ADD_NEW_ALGORITHM,
+  SUCCESS_UPDATE_ALGORITHM,
+} from 'utils/statusMessages';
+
+export const openAddDialog = (): easyPrimitiveActionReturn =>
+  easyPrimitiveAction({
+    type: DialogActions.OPEN_ADD_DIALOG,
+    // reducer ignores payload
+    payload: false,
+  });
+
+export const openChangeDialog = (
```

Comment:
Equally add a openAddDialog() action which will reset the dialog?

**At the moment, after you edited a algorithm and click on the add button again, the fields are not reset.**

Maybe generally delete all contents if an edit is canceled? We only want to keep the input for the add case I guess :)

Reply:
Of course, you are totally right. It should be a concern of the reducer.
Fix in commit 3755b26d